### ISSUE TITLE
[Bug]: username@hostname  can't transfor username and hostname

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#5179]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

CREATE USER daisy@192.168.1.10 IDENTIFIED BY '123456';
CREATE USER 'neil@192.168.1.10' IDENTIFIED BY '123456';
CREATE USER 'neil'@'192.168.1.10' IDENTIFIED BY '123456';
![image](https://user-images.githubusercontent.com/113406637/191649039-78662772-e50a-4c5c-a13a-811082def562.png)
![image](https://user-images.githubusercontent.com/113406637/191649063-67cfcb5c-ddb0-4df2-a832-914536800ee4.png)

**Comment by daviszhen** [comment]
the results are same as mysql:
create user 'neil@192.168.1.10' identified by '123456';

**Comment by daviszhen** [comment]
it is bug:
create user daisy@192.168.1.10 identified by '123456';

**Comment by daviszhen** [comment]
it was fixed at #5869

**Comment by heni02** [comment]
confirm,closed
![image](https://user-images.githubusercontent.com/113406637/197669688-820d5b8b-4416-4552-8fef-076fc83682e7.png)
